src/main/webapp/app/shared/subject/subject-dialog.component.ts

```diff
@@ -88,8 +92,8 @@ export class SubjectDialogComponent implements OnInit, OnDestroy {
             this.dateOfBirth = this.formatter.parse(this.subject.dateOfBirth.toString());
         }
 
-       this.delusion1$ = this.subject.attributes?.delusion_1 ?? null;
-       this.delusion2$ = this.subject.attributes?.delusion_2 ?? null;
+       this.delusion1  = this.subject.attributes?.delusion_1 ? this.delusions.find((d=>d.label=this.subject.attributes.delusion_1)):this.delusions[12];
```

Comment:
when initialising these -> you need to copy an object {...object} 

Also for find d.label you only use ommitted "==/===" 


In case like this put LOTS of console logs before and after and see what happens - thhat way you would be able to see the find(=) does not work properly and then that the changing of delusion1 influences delusion2 because the object was not copied